pkg/controllers/provisioning/scheduling/node.go

```diff
@@ -27,65 +27,133 @@ import (
 // Node is a set of constraints, compatible pods, and possible instance types that could fulfill these constraints. This
 // will be turned into one or more actual node instances within the cluster after bin packing.
 type Node struct {
-	Constraints         *v1alpha5.Constraints
+	v1alpha5.Constraints
 	InstanceTypeOptions []cloudprovider.InstanceType
 	Pods                []*v1.Pod
-	unschedulable       bool // if true, the node will accept no more pods
+
+	podResources    v1.ResourceList
+	daemonResources v1.ResourceList
+	daemons         []daemonInfo
+}
+
+type daemonInfo struct {
```

Comment:
Optional naming/style. I lean away from abstractions like `data`, `info`, `manager`, etc, since they tend to not convey additional meaning. In this case, I think we can get away with just calling it a `Daemon`.